### ISSUE TITLE
Swallowed bootstrap() failure leaves plugin inert but invisible under failOnInternalError=true

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#1096]
## Summary

When the analyzed app's `bootstrap/app.php` is discovered and `Application::bootstrap()` throws partway through (e.g. a bad `config/*.php`), the plugin **catches and swallows** the error, returns a **partially-booted** app, and continues. Service providers never register, so the plugin runs effectively **inert** (no model metadata, no facade bindings) yet Psalm reports a clean, successful run — **even with `failOnInternalError=true`**. The degraded state is only observable via `bin/psalm-laravel diagnose`, never from a normal `psalm` invocation.

The net effect: a green run that silently lost most of the plugin's value. Easy to mistake for "the plugin found nothing to add."

## Mechanism

`src/Providers/ApplicationProvider.php`:

```php
// doGetApp(), ~lines 181-186
try {
    $consoleApp->bootstrap();          // throws here (e.g. TypeError in a config file)
} catch (\Throwable $bootstrapError) {
    self::$bootstrapError = $bootstrapError;   // stored, never rethrown
}
return $app;                           // returns a partial app, normally
```

Because the throwable is caught **inside** the plugin (one layer below the plugin/Psalm boundary), it never escapes a plugin event-handler into Psalm's hook dispatcher. `failOnInternalError` only governs that boundary, so it has nothing to act on. The bootstrap chain aborts before `RegisterProviders` / `BootProviders`, leaving the app with no providers.

The tolerate-partial-bootstrap design (lines ~176-180) is intentional and good for crash-resistance — one bad config file shouldn't disable the plugin for the whole run. The problem is purely **observability**: the failure is invisible unless you run `diagnose`.

## Real-world repro

`HDInnovations/UNIT3D` ships `config/secure-headers.php`:

```php
'https://'.parse_url(env('VITE_ECHO_ADDRESS'), PHP_URL_HOST)...
```

Under static analysis there is no `.env`, so `env('VITE_ECHO_ADDRESS')` is `null` and `parse_url(null)` throws `TypeError: Argument #1 ($url) must be of type string, null given` (PHP 8). `bootstrap()` aborts during `LoadConfiguration`; providers never boot. `diagnose` shows:

```
Boot mode
  Mode  real bootstrap/app.php discovered
  ! Bootstrap warning: parse_url(): Argument #1 ($url) must be of type string, null given
```

Observed impact on that project: ~40% more reported issues and ~8pp lower type-coverage vs. a healthy boot, because the plugin's Eloquent/facade magic never engages and stops suppressing false `Undefined*` / `Mixed*` findings.

## Proposed fix

Make a degraded boot **loud by default**, without abandoning crash-resistance:

1. When `getBootstrapError()` is non-null (or `getBootMode() === 'testbench_fallback'` for a project that has a real `bootstrap/app.php`), surface it from a **normal** `psalm` run — e.g. a single `PluginBootstrapDegraded`-style info/warning line naming the throwable, not just behind `diagnose`.
2. Optionally gate visibility/severity on `failOnInternalError`: when it is `true`, treat a swallowed bootstrap failure as a reportable internal error (the user explicitly asked to fail on internal errors — a half-booted app is arguably one), while keeping the current silent-degrade behavior when it is `false`.
3. Consider seeding common env defaults before `bootstrap()` so unset `env()` reads don't `TypeError` mid-config (reduces how often this path is hit at all).

## Acceptance

- A normal `psalm` run on a project whose `bootstrap()` throws emits a visible signal that the plugin is degraded (no need to run `diagnose`).
- With `failOnInternalError=true`, a swallowed bootstrap failure is reportable rather than invisible.
- Crash-resistance is preserved: a single bad config file still does not abort the whole analysis when the user has not opted into failing.

## Related

- #766 (silent Testbench/degraded-boot fallback)
